app/main.py

```diff
@@ -1,6 +1,27 @@
-from typing import Callable
+from typing import Callable, List
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def wrapper(*args: Callable) -> object:
+        if args in cache_storage:
+            print("Getting from cache")
+            return cache_storage[args]
+        else:
+            print("Calculating new result")
```

Comment:
you don't need else because if has return, it means it'll not reach the code after it